File: AAS.Domain/Bids/BidBlank.cs
using AAS.Domain.Bids.Enums;
using AAS.Tools.Types.IDs;

namespace AAS.Domain.Bids;

public class BidBlank
{
    public ID? Id { get; set; }
    public Int32? Number { get; set; }
    public string? Title { get; set; }
    public string? Description { get; set; }
    public string? DenyDescription { get; set; }
    public BidStatus Status { get; set; }
    public DateOnly? ApproximateDate { get; set; }

    public BidBlank(ID? id, Int32? number, string? title, string? description, string? denyDescription, BidStatus status, DateOnly? approximateDate)
    {
        Id = id;
        Number = number;
        Title = title;
        Description = description;
        DenyDescription = denyDescription;
        Status = status;
        ApproximateDate = approximateDate;
    }
}

File: AAS.BackOffice/client_app/src/domain/bids/bidBlank.ts
import { Bid } from "./bid"

export interface BidBlank {
    id: string | null,
    title: string | null,
    description: string | null,
    denyDescription: string | null,
    status: string | null,
    acceptanceDate: Date | null,
    approximateDate: Date | null,
}

export namespace BidBlank {
    export function getDefault(): BidBlank {
        return {
            id: null,
            title: null,
            description: null,
            denyDescription: null,
            status: null,
            acceptanceDate: null,
            approximateDate: null,
        }
    }

    export function fromBid(bid: Bid): BidBlank {
        return {
            id: bid.id,
            title: bid.title,
            description: bid.description,
            denyDescription: bid.deynDescription,
            status: bid.status,
            acceptanceDate: bid.acceptanceDate,
            approximateDate: bid.approxmateDate,
        }
    }
}